### PR TITLE
refactor(Dialog):slot exit judgment

### PR DESCRIPTION
**Pull request by xincheng-1999**
Before submitting a pull request, please read the [contributing guide](https://vant-contrib.gitee.io/vant/#/en-US/contribution).

在提交 pull request 之前，请阅读 [贡献指南](https://vant-contrib.gitee.io/vant/#/zh-CN/contribution)。

### CODE REVIEW

**Comment by inottn** [comment]
what problem does this PR resolve?

File: packages/vant/src/dialog/Dialog.tsx [hunk 1]
```diff
@@ -170,7 +170,7 @@ export default defineComponent({
     };
 
     const renderContent = () => {
-      if (slots.default) {
+      if (slots.default?.()) {
```
Comment (chenjiahan) [inline packages/vant/src/dialog/Dialog.tsx:173]: This change will cause the render function to run twice

Comment (xincheng-1999) [reply packages/vant/src/dialog/Dialog.tsx:173]: `  <van-dialog
    v-model:show="refShowDialog"
    class="g-dialog--b"
    :title="title"
    :message="message"
    :confirm-button-text="confirmButtonText"
    :show-cancel-button="true"
    :cancel-button-text="cancelButtonText"
    :width="px2remWithUnit(686)"
    :overlay-style="{ backgroundColor: bgTransparent ? 'transparent' : generateBlackTransparency(0.4) }"
    :cancel-button-disabled="cancelButtonDisabled"
    :confirm-button-disabled="confirmButtonDisabled"
    @confirm="handleConfirm"
    @cancel="handleCancel"
  >
    <slot />
  </van-dialog>`
  如果组件二次封装希望自由选择渲染message或者是slot，这个<slot />标签会导致dialog内部判断slots.default永远是true，这样即使外层不传入slot,也不会走渲染message的逻辑

Comment (zhousg) [reply packages/vant/src/dialog/Dialog.tsx:173]: <slot v-if="!message && $slots.default" />    应该好使

Comment (xincheng-1999) [reply packages/vant/src/dialog/Dialog.tsx:173]: > This change will cause the render function to run twice

这个没考虑到应该用一个变量接收

Comment (zhousg) [reply packages/vant/src/dialog/Dialog.tsx:173]: 是的

**Pull request closed without merging**